File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, k: int, nums: List[int]) -> int:
        i = 0 
        j = 0 

        min_len = float('inf') 
        max_sum = 0
        
        while j < len(nums):
            max_sum += nums[j]

            if max_sum < k:
                j += 1
            elif max_sum >= k:
                min_len = min(min_len, (j - i + 1))

                while max_sum >= k:
                    min_len = min(min_len, (j - i + 1))
                    max_sum -= nums[i]
                    i += 1

                j += 1

        return 0 if min_len == float('inf') else min_len 

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(nums: number[]): number {
        let answer :number = Number.MAX_VALUE

        let left = 0;
        let right = nums.length - 1;

        while (left <= right) {
            let mid = Math.floor((left + right) / 2);

            if (nums[left] <= nums[right]) {
                answer =  Math.min(answer,nums[left])
                break
            }

            if (nums[left] <= nums[mid]) {
                answer = Math.min(nums[mid], nums[left])
                left = mid + 1
            } else {
                answer = Math.min(answer,nums[mid])
                right = mid - 1
            }
        }
        return answer;
    }
}
